### PR TITLE
Added Visual Basic tuple documentation

### PR DESCRIPTION
**Pull request by rpetrusha**
# Added Visual Basic tuple documentation

## Suggested Reviewers

@BillWagner @AnthonyDGreen

### CODE REVIEW

File: docs/visual-basic/programming-guide/language-features/data-types/tuples.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+| CompareTo | Compares the current tuple to another tuple with the same number of elements. |
+| Equals | Determines whether the current tuple is equal to another tuple or object. |
+| GetHashCode | Calculates the hash code for the current instance. |
+| ToString | Returns the string representation of this tuple, which takes the form `(Item1, Item2...)`, where `Item1` and `Item2` represent the values of the tuple's fields. |
+
+In addition, the **ValueTuple** types implement <xref:System.Collections.IStructuralComparable> and <xref:System.Collections.IStructurealEquatable> interfaces, which allow you to define customer comparers.
+
+## Assignment and tuples
+
+Visual Basic supports assignment between tuple types that have the same number of fields and the same type for each field. Those types must be exact compile-time matches. Other conversions are not considered for assignments. Let's look at the kinds of assignments that are allowed between tuple types.
```
Comment (BillWagner) [inline docs/visual-basic/programming-guide/language-features/data-types/tuples.md]: The second sentence is incorrect. The types of each field don't have to be the same. There must be an implicit conversion from the field type in the right-hand-side tuple to the field type in the left-hand-side tuple.  As a trivial example:

```vb
        Dim longs = (19L, 23L)
        Dim ints = (19, 23)
        longs = ints
```

Comment (BillWagner) [inline docs/visual-basic/programming-guide/language-features/data-types/tuples.md]: And, of course, I made the same mistake in the C# version.  See PR #2027

File: docs/visual-basic/programming-guide/language-features/data-types/tuples.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+We can return a tuple from the parsing operation if we wrap the call to the <xref:System.Int32.TryParse%2A> method in our own method. In the following example, `NumericLibrary.ParseInteger` calls the <xref:System.Int32.TryParse%2A> method and returns a named tuple with two elements. 
+
+[!code-vb[Return](../../../../../samples\snippets\visualbasic\programming-guide\language-features\data-types/tuple-return.vb#2)]
+
+You can then call the method with code like the following:
+
+[!code-vb[Return](../../../../../samples\snippets\visualbasic\programming-guide\language-features\data-types/tuple-return.vb#2)]
+
+## Visual Basic tuples and tuples in the .NET Framework
```
Comment (BillWagner) [inline docs/visual-basic/programming-guide/language-features/data-types/tuples.md]: Consider updating the descriptions to compare `System.Tuple` and `System.ValueTuple`. The two types are different. The VB (and C#) teams provided language features that map to System.Tuple.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Overall, this looks great.

I pointed out a couple errors (one of which I'd done in the C# topic tool 😞  I fixed that one in PR #2027, so this helped find that error.

File: docs/visual-basic/programming-guide/language-features/data-types/tuples.md (hunk 2 above)
Comment (rpetrusha) [inline docs/visual-basic/programming-guide/language-features/data-types/tuples.md]: @BillWagner, language features map to ValueTuple, no?

Comment (rpetrusha) [inline docs/visual-basic/programming-guide/language-features/data-types/tuples.md]: @BillWagner, I've addressed your comments and modified the example slightly.

Comment (BillWagner) [inline docs/visual-basic/programming-guide/language-features/data-types/tuples.md]: Yes, Ron you are correct ValueTuple, not Tuple

**Review by BillWagner — APPROVED** [review]

**Pull request merged**